doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,394 @@
+# Solve an equation algebraically
+
+Use SymPy to solve an equation algebraically (symbolically). For example, solving $x^2 = y$ yields $x \in \{-\sqrt{y},\sqrt{y}\}$.
+
+Alternatives to consider:
+- SymPy can also [solve many other types of problems including sets of equations](index.md).
+- Some equations cannot be solved algebraically (either at all or by SymPy), 
+so you may have to {func}`solve your equation numerically <sympy.solvers.solvers.nsolve>` instead.
+
+There are two high-level functions to solve equations, {func}`~.solve` and {func}`~.solveset`.
+Here is a simple example of each:
+
+{func}`~.solve`
+
+```py
+>>> from sympy.abc import x, y
+>>> from sympy import solve
+>>> solution = solve(x ** 2 - y, x, dict=True)
+>>> print(solution)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+```
+
+{func}`~.solveset`
+
+```py
+>>> from sympy import solveset
+>>> from sympy.abc import x, y
+>>> solution = solveset(x**2 - y, x)
+>>> print(solution)
+{-sqrt(y), sqrt(y)}
+```
+
+Here are recommendations on when to use:
+
+- {func}`~.solve`
+    - You need to programmatically extract components (expressions, or individual symbols or constants) 
+    from the output by traversing it.
+    - You want an explicit solution expression that you can use to substitute into something else.
+    - You want to apply assumptions to symbols (for example, $x$ is real) to determine the 
+    domain of the solutions.
+
+- {func}`~.solveset`
+    - You want a consistent input and output interface.
+    - You want to get all the solutions, including if there are infinitely many.
+    - You want to limit the domain of the solutions (for example, to real values) directly.
```

Comment:
I think many users would say that the output of `solveset` is not consistent. Really they want a FiniteSet but instead they have a Complement of an Intersection of a Union etc: how do I get the values out of this?

Also the part about limiting the domain of the solutions to the reals doesn't make much sense since solve also allows you to do that. What solveset allows you to do is to limit the domain of the solutions to any arbitrary set e.g.:
```python
In [1]: solveset(sin(x), x, Interval(-pi, pi))
Out[1]: {0, -π, π}
```